src/databases/transaction_database.cpp

```diff
@@ -421,45 +424,60 @@ bool transaction_database::candidize(link_type link, bool candidate)
 // Confirm/Unconfirm.
 // ----------------------------------------------------------------------------
 
-bool transaction_database::confirm(const transaction::list& transactions,
-    size_t height, uint32_t median_time_past)
+// This does not add to the output cache since it would require reading the tx.
+// This also does not confirm unconfirmed cached outputs (should not exist).
+// This also does not remove cached items on unconfirmation (should not call).
+bool transaction_database::confirm(file_offset link, size_t height,
+    uint32_t median_time_past, size_t position)
 {
-    uint32_t position = 0;
-    for (const auto& tx: transactions)
-        if (!confirm(tx.metadata.link, height, median_time_past, position++))
+    // Avoid population of prevouts for coinbase (confirmation optimization).
+    if (position != 0)
```

Comment:
@evoskuil this has an implication on the on the `transaction_result::is_spent` method. We will need to skip the spentness check on the 0th output.

I came across it when trying to bring back the commented out test for confirming a list of transactions; which has now changed to confirming transactions in a block.

Do we need this test? I ask, because in the confirmed_spend mehtod we anyway check if point is null.

Reply:
Actually, ignore all of the above. It is the tests that needed to be fixed a bit. I'll send a PR for your fork with the tests fixed.